### DISCUSSION TITLE
computing parent orientations of each pixel

### DISCUSSION THREAD

**Discussion by nikozol** [mtex-toolbox/mtex#591] [Ask Anything]
Hi colleagues, 
Using the scripts, which take place in the section “Martensite Parent Grain Reconstruction” of Documentation, I have carried out successfully parent austenite grain reconstruction. The error appeares when computing parent orientations of each pixel in EBSD:
[parentEBSD(isNowFCC).orientations, fit] = calcParent(ebsd(isNowFCC).orientations,...
  parentGrains(parentEBSD(isNowFCC).grainId).meanOrientation,fcc2bcc);

Error message:
Error using orientation/dot_outer (line 25)
comparing orientations of different phase not yet supported

Error in quaternion/angle_outer (line 15)
omega = real(2*acos(abs(dot_outer(q1,q2,varargin{:}))));

Error in calcParent (line 60)
  fit = angle_outer(mori,p2c,'noSym2');

Error in OR_Parent_mtextools (line 217)
[parentEBSD(isNowFCC).orientations, fit] = calcParent(ebsd(isNowFCC).orientations,...
______________

A distinction of my data from those in Documentation - only alpha-phase takes place: 
% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('432', [2.9 2.9 2.9], 'mineral', 'Iron - Alpha', 'color', [0.53 0.81 0.98])};
After excluding non-indexed points:
ebsd = EBSD (show methods, plot)
  Phase   Orientations       Mineral         Color  Symmetry  Crystal reference frame
     0  394212 (100%)  Iron - Alpha  LightSkyBlue       432 

Since I have not austenite in my data, I set symmetry when computing orientation as follows:
%% OR determination
csBCC = ebsd.CSList{2}; % martensite bcc
csFCC = ebsd.CSList{2}; % austenite fcc

Nikolai

**Comment by AmitS07** [comment]
Hello All, 
I am also getting an error while **Computing Child Variants**. The script line which is giving error is as following-

% consider only martensite pixels that now belong to austenite grains
isNowFCC = parentGrains.phaseId(max(1,parentEBSD.grainId)) == 3 & parentEBSD.phaseId == 2;

The Error:

**Index exceeds array bounds**

Error in grain2d/subsref (line 44)
  [varargout{1:nargout}] = builtin('subsref',grains,s);


Could any one address this issue? 

Thanks in advance

-----
Amit
